File: frontend/src/app/shared/ActiveDirectorySearchRequestDto.ts
import { ActiveDirectoryConnectionDataDto } from './ActiveDirectoryConnectionDataDto'

export class ActiveDirectorySearchRequestDto {
  connectionData: ActiveDirectoryConnectionDataDto
  baseSearch: string
  filterQuery: string
  userFieldToActiveDirectoryAttributeMapping: any


  constructor(connectionData: ActiveDirectoryConnectionDataDto = new ActiveDirectoryConnectionDataDto(),
              baseSearch: string                               = '',
              filterQuery: string                              = '',
              userFieldToActiveDirectoryAttributeMapping       = {}) {
    this.connectionData = connectionData
    this.baseSearch = baseSearch
    this.filterQuery = filterQuery
    this.userFieldToActiveDirectoryAttributeMapping = userFieldToActiveDirectoryAttributeMapping

    this.connectionData.host = 'ldap.forumsys.com'
    this.connectionData.port = 389
    this.connectionData.distinguishedName = 'cn=read-only-admin,dc=example,dc=com'
    this.connectionData.password = 'password'

    this.baseSearch = 'dc=example,dc=com'
    this.filterQuery = 'objectClass=person'
  }
}

File: backend/src2/SharedModels/DTOs/ActiveDirectorySearchRequestDto.cs
using System.Collections.Generic;

namespace SharedModels.DTOs
{
    public class ActiveDirectorySearchRequestDto
    {
        public ActiveDirectoryConnectionDataDto ConnectionData { get; set; }
        public string BaseSearch { get; set; }
        public string FilterQuery { get; set; }
        public Dictionary<string, string> UserFieldToActiveDirectoryAttributeMapping { get; set; }
    }

    public class ActiveDirectoryConnectionDataDto
    {
        public string Host { get; set; }
        public int Port { get; set; }
        public string DistinguishedName { get; set; }
        public string Password { get; set; }
    }
}